File: src/bin/day1_1.rs
use std::fs::read_to_string;

const IS_PROD: bool = true;

const TEST_INPUT_PATH: &str = "./inputs/day1.test.in";
const PROD_INPUT_PATH: &str = "./inputs/day1.prod.in";
const INPUT_PATH: &str = if IS_PROD {
    PROD_INPUT_PATH
} else {
    TEST_INPUT_PATH
};

fn main() {
    let mut a: Vec<usize> = vec![];
    let mut b: Vec<usize> = vec![];

    read_to_string(INPUT_PATH)
        .unwrap()
        .trim()
        .split("\n")
        .for_each(|l| {
            let values = l
                .split("   ")
                .map(|v| v.parse::<usize>().unwrap())
                .collect::<Vec<usize>>();

            a.push(*values.get(0).unwrap());
            b.push(*values.get(1).unwrap());
        });

    a.sort();
    b.sort();

    let mut result = 0;

    for n in 0..a.len() {
        let va = a.get(n).unwrap();
        let vb = b.get(n).unwrap();

        result += va.abs_diff(*vb);
    }

    println!("Result : {:?}", result);
}

File: src/ts/day1_1.ts
import path from "node:path";

const IS_PROD = true;
const TEST_INPUT_PATH = path.join(__dirname, "../../inputs/day1.test.in");
const PROD_INPUT_PATH = path.join(__dirname, "../../inputs/day1.prod.in");
const INPUT_PATH = IS_PROD ? PROD_INPUT_PATH : TEST_INPUT_PATH;

const file = Bun.file(INPUT_PATH);
const text = await file.text();

const a: number[] = [];
const b: number[] = [];

text.trim().split("\n").forEach((line) => {
  const [va, vb] = line.split("   ");
  a.push(Number.parseInt(va, 10));
  b.push(Number.parseInt(vb, 10));
});

a.sort((a, b) => a - b);
b.sort((a, b) => a - b);

let result = 0;

for (let i = 0; i < a.length; i += 1) {
  result += Math.abs(a[i] - b[i]);
}

console.log("Result :", result);
